pkg/admission/clusterworkspacetypeexists/admission.go

```diff
@@ -97,25 +98,29 @@ func (o *clusterWorkspaceTypeExists) Admit(ctx context.Context, a admission.Attr
 		return admission.NewForbidden(a, fmt.Errorf("not yet ready to handle request"))
 	}
 
-	clusterName, err := genericapirequest.ClusterNameFrom(ctx)
-	if err != nil {
-		return apierrors.NewInternalError(err)
-	}
-
-	resolveCwt := func() (*tenancyv1alpha1.ClusterWorkspaceType, error) {
-		cwt, err := o.typeLister.Get(clusters.ToClusterAwareKey(clusterName, strings.ToLower(cw.Spec.Type)))
-		if err != nil && apierrors.IsNotFound(err) {
-			return nil, fmt.Errorf("spec.type %q does not exist", cw.Spec.Type)
-		} else if err != nil {
-			return nil, err
-		}
-		return cwt, nil
-	}
-
 	if a.GetOperation() == admission.Create {
-		cwt, err := resolveCwt()
-		if err != nil {
-			return admission.NewForbidden(a, err)
+		// if the user has not provided any type, use the default from the parent workspace
+		empty := tenancyv1alpha1.ClusterWorkspaceTypeReference{}
+		if cw.Spec.Type == empty {
+			clusterName, err := genericapirequest.ClusterNameFrom(ctx)
+			if err != nil {
+				return apierrors.NewInternalError(err)
+			}
+			parentCwt, resolutionError := o.resolveParentClusterWorkspaceType(clusterName)
+			if resolutionError != nil {
+				return admission.NewForbidden(a, resolutionError)
+			}
+			if parentCwt == nil {
+				return admission.NewForbidden(a, errors.New("spec.type must be set in the root logical cluster"))
```

Comment:
avoid "logical cluster" in everything user-facing. Here it is the root workspace.

Reply:
unresolved